feat: new introductory javascript project (#54305)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-pyramid-generator/660f0c34aad72dc712b97624.md

---
id: 660f0c34aad72dc712b97624
title: Step 22
challengeType: 0
dashedName: step-22
---

# --description--

> The other method you will need to know for this project is the `.pop()` method. This method "pops" the last value from the array and <dfn>returns</dfn> it.

Reviewer: ```suggestion
Another method essential for this project is the `.pop()` method. It removes the last element from an array and <dfn>returns</dfn> that element.
```

> When a method returns a value, you can think of it as giving the value back to you so you can use it elsewhere.

Reviewer: ```suggestion
When a method returns a value, you can think of it as giving the value back to you, making it available for use in other parts of your code.
```

Declare a `popped` variable, and assign it the result of `rows.pop()`. Then, log your `popped` variable.

# --hints--

You should declare a `popped` variable.

```js
assert.match(code, /popped/);
```

You should use `let` to declare your `popped` variable.

```js
assert.match(code, /let\s+popped/);
```

You should call the `.pop()` method.

```js
assert.match(code, /\.pop\(\s*\)/);
```

> You should call the `.pop()` method on your `rows` array. Don't forget your semi-colon.
> 
> ```js
> assert.match(code, /rows\.pop\(\s*\)/);
> ```
> 
> You should log your `popped` variable. Don't forget your semi-colon.
> 
> ```js
> assert.match(code, /console\.log\(\s*popped\s*\)/);
> ```

Reviewer 2: This doesn't look like it checks for semi colons. But they should be optional anyway

````suggestion
You should call the `.pop()` method on your `rows` array. 

```js
assert.match(code, /rows\.pop\(\s*\)/);
```

You should log your `popped` variable. 

```js
assert.match(code, /console\.log\(\s*popped\s*\)/);
```
````
